builtin/providers/google/resource_dataflow_job_test.go

```diff
@@ -0,0 +1,70 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataflowJobCreate(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDataflowJobDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccDataflowJob,
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataflowJobExists(
+						"google_dataflow_job.big_data"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDataflowJobDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_dataflow_job" {
+			continue
+		}
+
+		config := testAccProvider.Meta().(*Config)
+		job, _ := config.clientDataflow.Projects.Jobs.Get(config.Project, rs.Primary.ID).Do()
+		if job != nil {
+			return fmt.Errorf("Job still present")
+		}
+	}
+
+	return nil
+}
+
+func testAccDataflowJobExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+		config := testAccProvider.Meta().(*Config)
+		_, err := config.clientDataflow.Projects.Jobs.Get(config.Project, rs.Primary.ID).Do()
+		if err != nil {
+			return fmt.Errorf("Job does not exist")
+		}
+
+		return nil
+	}
+}
+
+var testAccDataflowJob = fmt.Sprintf(`
```

Comment:
Okay, I've got the tests running locally. The main issue as far as I can see is that the notion of "deletion" doesn't align perfectly well with Dataflow's concept of execution state – specifically, "deleting" a job means stopping it (either cancelling or draining), but the job is still there.

Maybe instead of using the job id as the resource id we should use the job name – which needs to be unique within a project. Then we can consider the job "deleted" if it's not running or in the process of cancelling or draining. Basically, once Delete returns, it should be possible to Create the job anew, which implies that there mustn't be a name conflict.

@danawillow what do you think?